rust/telemetry/src/feature_flags.rs

```diff
@@ -0,0 +1,144 @@
+use std::{sync::LazyLock, time::Duration};
+
+use anyhow::{bail, Context as _, Result};
+use parking_lot::RwLock;
+use serde::{Deserialize, Serialize};
+use tokio::runtime::Runtime;
+
+const POSTHOG_API_KEY: &str = "phc_uXXl56plyvIBHj81WwXBLtdPElIRbm7keRTdUCmk8ll";
+const RE_EVAL_DURATION: Duration = Duration::from_secs(5 * 60);
+
+static RUNTIME: LazyLock<Runtime> = LazyLock::new(init_runtime);
+
+// Process-wide storage of enabled feature flags.
+//
+// Defaults to everything off.
+static FEATURE_FLAGS: LazyLock<RwLock<FeatureFlags>> = LazyLock::new(RwLock::default);
+
+pub fn icmp_unreachable_instead_of_nat64() -> bool {
+    FEATURE_FLAGS.read().icmp_unreachable_instead_of_nat64
+}
+
+pub fn drop_llmnr_nxdomain_responses() -> bool {
+    FEATURE_FLAGS.read().drop_llmnr_nxdomain_responses
+}
+
+pub(crate) fn reevaluate(user_id: String) {
+    RUNTIME.spawn(async move {
+        let flags = decide(user_id)
+            .await
+            .inspect_err(|e| tracing::debug!("Failed to evaluate feature flags: {e:#}"))
+            .unwrap_or_default();
+
+        tracing::debug!(?flags, "Evaluated feature-flags");
+
+        *FEATURE_FLAGS.write() = flags;
+
+        sentry::Hub::main().configure_scope(|scope| {
+            scope.set_context("flags", sentry_flag_context(flags));
+        });
+    });
+}
+
+/// Initialize the runtime to use for evaluating feature flags.
+fn init_runtime() -> Runtime {
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .worker_threads(1) // We only need 1 worker thread.
+        .thread_name("feature-flag-worker")
+        .enable_io()
+        .enable_time()
+        .build()
+        .expect("to be able to build runtime");
+
+    runtime.spawn(async move {
+        loop {
+            tokio::time::sleep(RE_EVAL_DURATION).await;
+
+            let Some(user_id) = sentry::Hub::main()
+                .configure_scope(|scope| scope.user().and_then(|u| u.id.clone()))
+            else {
+                continue; // Nothing to do if we don't have a user-id set.
+            };
+
+            reevaluate(user_id);
+        }
+    });
+
+    runtime
+}
+
+async fn decide(distinct_id: String) -> Result<FeatureFlags> {
+    let response = reqwest::ClientBuilder::new()
+        .connection_verbose(true)
+        .build()?
+        .post("https://us.i.posthog.com/decide?v=3")
+        .json(&DecideRequest {
+            api_key: POSTHOG_API_KEY.to_string(),
+            distinct_id,
+        })
+        .send()
+        .await
+        .context("Failed to send POST request")?;
+
+    let status = response.status();
+
+    if !status.is_success() {
+        let body = response.text().await.unwrap_or_default();
+
+        bail!("Failed to get feature flags; status={status}, body={body}")
+    }
+
+    let json = response
+        .json::<DecideResponse>()
+        .await
+        .context("Failed to deserialize response")?;
```

Comment:
Does this log the response that failed? Just thinking we'll probably get occasional failures from both Posthog and possibly any filtering being done in the user's environment.

Reply:
It doesn't no, just any deserialisation errors.